java/code/src/com/redhat/rhn/frontend/xmlrpc/contentmgmt/ContentManagementHandler.java

```diff
@@ -836,4 +877,45 @@ public int promoteProject(User loggedInUser, String projectLabel, String envLabe
         }
         return 1;
     }
+
+    /**
+     * Get a {@link ContentProject} instance using the project label
+     *
+     * @param loggedInUser the user
+     * @param projectLabel the project label
+     * @return the content project instance
+     * @throws EntityNotExistsFaultException when project doesn't exist
+     */
+    private ContentProject getContentProject(User loggedInUser, String projectLabel)
+            throws EntityNotExistsFaultException {
+        return ContentManager.lookupProject(projectLabel, loggedInUser)
+                .orElseThrow(() -> new EntityNotExistsFaultException(new EntityNotExistsException(projectLabel)));
+    }
+
+    /**
+     * Validates a content project for build/promote
+     *
+     * The validation fails only in case of an error. Info and warning messages are ignored since the build/promote
+     * operation can still be performed.
+     *
+     * @param project the {@link ContentProject} instance
+     * @throws ContentValidationFaultException when validation fails with messages
+     */
+    private void validateContentProject(ContentProject project) throws ContentValidationFaultException {
+        ContentProjectValidator projectValidator = new ContentProjectValidator(project,
+                Arrays.asList(
+                    new ModularSourcesValidator(),
+                    new ModularDependencyValidator(contentManager.getModulemdApi())
+                ));
```

Comment:
In the UI we seem to also have a project validator that consists of `ModularSourceValidator` and `ModularDependencyValidator`. Could some portion (at least the list of used validators) of the validator initialization be extracted to a common place, so that next time we want to add a validator, we just change it on one place?

Reply:
I'll use another constructor that adds all validators by default